src/sensors/softfusion/drivers/bmi270.h

```diff
@@ -398,15 +398,18 @@ struct BMI270 {
 
 		for (uint32_t i = 0u; i < bytes_to_read;) {
 			const uint8_t header = getFromFifo<uint8_t>(i, read_buffer);
-			if ((header & Fifo::ModeMask) == Fifo::SkipFrame
-				&& (i - bytes_to_read) >= 1) {
+			if ((header & Fifo::ModeMask) == Fifo::SkipFrame) {
+				if (i + 1 > bytes_to_read) {
```

Comment:
Couldn't we just do this? Also is this even necessary considering the condition of the for loop or does `getFromFifo` progress `i` (I believe this to be the case)?
```suggestion
				if (i >= bytes_to_read) {
```

Reply:
It's just coincidence that when you need to read 1 byte, you can use either `+ 1 > bytes_to_read` or `>= bytes_to_read`. In general, when you need to read X bytes, you can only use `+ X > bytes_to_read`. I chose this so that the "has enough bytes" check is consistent across all the cases.

The for loop is insufficient because you might end up reading past the end of the buffer.